### PR TITLE
certManager: support string array and string in caPool property

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

**Review by ondrejtomcik — APPROVED** [review]
Please update documentation of config properties in doc. it's important for auth.caPool

File: pkg/security/certManager/client/certManager_test.go [hunk 1]
```diff
@@ -45,6 +46,93 @@ AnQ1eTGKTGLdAZsV+NnZPL17nit1cbiN2g==
...
+	}{
+		{
+			name: "caPool as string",
+			args: args{
+				configYaml: `caPool: /tmp/test3570354545/ca3202122454
+keyFile: /tmp/test3570354545/key3658110735
+certFile: /tmp/test3570354545/crt4065348335
+`,
+			},
+			want: client.Config{
```
Comment (Danielius1922) [inline pkg/security/certManager/client/certManager_test.go:68]: tt.want is unused as far as I can see

File: pkg/security/certManager/client/certManager.go [hunk 2]
```diff
@@ -7,18 +7,26 @@ import (
...
 type Config struct {
-	CAPool          string `yaml:"caPool" json:"caPool" description:"file path to the root certificates in PEM format"`
-	KeyFile         string `yaml:"keyFile" json:"keyFile" description:"file name of private key in PEM format"`
-	CertFile        string `yaml:"certFile" json:"certFile" description:"file name of certificate in PEM format"`
-	UseSystemCAPool bool   `yaml:"useSystemCAPool" json:"useSystemCaPool" description:"use system certification pool"`
+	CAPool          interface{} `yaml:"caPool" json:"caPool" description:"file path to the root certificates in PEM format"`
+	KeyFile         string      `yaml:"keyFile" json:"keyFile" description:"file name of private key in PEM format"`
+	CertFile        string      `yaml:"certFile" json:"certFile" description:"file name of certificate in PEM format"`
+	UseSystemCAPool bool        `yaml:"useSystemCAPool" json:"useSystemCaPool" description:"use system certification pool"`
+	CAPoolArray     []string    `yaml:"-" json:"-"`
```
Comment (Danielius1922) [inline pkg/security/certManager/client/certManager.go]: I don't like that this is exported, this is an internal field generated from the `CAPool` field.

What about something like this:

```go
type Config struct {
...
  caPoolArray []string `yaml:"-" json:"-"`...
  validated bool
}

func (c *Config) CAPoolArray()([]string, error) {
  if !c.validated {
    return nil, error
  }
  return c.caPoolArray, nil
}
```

File: pkg/security/certManager/client/certManager_test.go (hunk 1 above)
Comment (jkralik) [reply pkg/security/certManager/client/certManager_test.go:68]: fixed :)

File: pkg/security/certManager/general/certManager_test.go [hunk 3]
```diff
@@ -113,15 +113,15 @@ func createTmpCertFiles(t *testing.T, caFile, crtFile, keyFile string) general.C
...
+		CAPool:   []string{caFile},
 		KeyFile:  keyFile,
 		CertFile: crtFile,
 	}
 	return cfg
 }
 
 func deleteTmpCertFiles(t *testing.T, cfg general.Config) {
-	err := os.Remove(cfg.CAPool)
+	err := os.Remove(cfg.CAPool[0])
```
Comment (Danielius1922) [inline pkg/security/certManager/general/certManager_test.go]: I would range over the array here, that way the code will continue to work if we ever modify the constructor to create CAPool with multiple files.

Comment (jkralik) [reply pkg/security/certManager/general/certManager_test.go]: fixed

File: pkg/security/certManager/client/certManager.go (hunk 2 above)
Comment (jkralik) [reply pkg/security/certManager/client/certManager.go]: fixed

Comment (Danielius1922) [reply pkg/security/certManager/client/certManager.go]: yep, good job.

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**